maps/src/test/java/org/odk/collect/maps/layers/OfflineMapLayersPickerTest.kt

```diff
@@ -361,12 +367,151 @@ class OfflineMapLayersPickerTest {
+    }
+
+    @Test
+    fun `clicking delete shows the confirmation dialog`() {
+        whenever(referenceLayerRepository.getAll()).thenReturn(listOf(
+            ReferenceLayer("1", TempFiles.createTempFile(), "layer1")
+        ))
+
+        launchFragment()
+
+        scheduler.flush()
+
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.arrow)).perform(click())
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.delete_layer)).perform(scrollTo(), click())
+
+        onView(withText(string.cancel)).inRoot(isDialog()).check(matches(isDisplayed()))
+        onView(withText(string.delete_layer)).inRoot(isDialog()).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun `clicking delete and canceling does not remove the layer`() {
+        val layerFile1 = TempFiles.createTempFile()
+        whenever(referenceLayerRepository.getAll()).thenReturn(listOf(
+            ReferenceLayer("1", layerFile1, "layer1")
+        ))
+
+        launchFragment()
+
+        scheduler.flush()
+
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.arrow)).perform(click())
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.delete_layer)).perform(scrollTo(), click())
+
+        onView(withText(string.cancel)).inRoot(isDialog()).perform(click())
+
+        onView(withId(R.id.layers)).check(matches(RecyclerViewMatcher.withListSize(2)))
+        onView(withId(R.id.layers)).perform(scrollToPosition<RecyclerView.ViewHolder>(0))
+        onView(withRecyclerView(R.id.layers).atPositionOnView(0, R.id.title)).check(matches(withText(string.none)))
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.title)).check(matches(withText("layer1")))
+        assertThat(layerFile1.exists(), equalTo(true))
+    }
+
+    @Test
+    fun `clicking delete and confirming removes the layer`() {
+        val layerFile1 = TempFiles.createTempFile()
+        val layerFile2 = TempFiles.createTempFile()
+        whenever(referenceLayerRepository.getAll()).thenReturn(listOf(
+            ReferenceLayer("1", layerFile1, "layer1"),
+            ReferenceLayer("2", layerFile2, "layer2")
+        ))
+
+        launchFragment()
+
+        scheduler.flush()
+
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.arrow)).perform(click())
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.delete_layer)).perform(scrollTo(), click())
+
+        onView(withText(string.delete_layer)).inRoot(isDialog()).perform(click())
+        scheduler.flush()
+
+        onView(withId(R.id.layers)).check(matches(RecyclerViewMatcher.withListSize(2)))
+        onView(withRecyclerView(R.id.layers).atPositionOnView(0, R.id.title)).check(matches(withText(string.none)))
+        onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.title)).check(matches(withText("layer2")))
+        assertThat(layerFile1.exists(), equalTo(false))
```

Comment:
This feels like it bleeds through the encapsulation of `ReferenceLayerRepository`. I think really there should be a `ReferenceLayerRepository#delete` and this should check that that's been called or that the layer is gone from fake.

Reply:
Fixed.